stl/src/filesystem.cpp

```diff
@@ -854,21 +824,32 @@ struct alignas(long long) _Aligned_file_attrs {
             _Flags, _Get_file_attributes_data)) { // caller wants something GetFileAttributesExW might provide
         if (_Symlink_attribute_hint == __std_fs_file_attr::_Invalid
             || !_Bitmask_includes(_Symlink_attribute_hint, __std_fs_file_attr::_Reparse_point)
-            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so GetFileAttributesExW
+            || !_Follow_symlinks) { // we might not be a symlink or not following symlinks, so FindFirstFileW
                                     // would return the right answer
-            _Aligned_file_attrs _Aligned_attrs;
-            auto& _Data = _Aligned_attrs._Data;
-            if (!GetFileAttributesExW(_Path, GetFileExInfoStandard, &_Data)) {
-                return __std_win_error{GetLastError()};
+
+            // check for file names that contain `?` or `*` (globbing characters)
+            // these are invalid file names, and will give us the wrong answer with `FindFirstFile`

```

Comment:
The codebase isn't completely consistent about this, but I recommend following a convention that if a comment is multiple sentences, we should use capitalization and punctuation to ensure readability.

Additionally, it is helpful to mention function names exactly, so that they're searchable:
```suggestion
            // Check for file names that contain `?` or `*` (globbing characters).
            // These are invalid file names, and will give us the wrong answer with `FindFirstFileW`.
```